### ISSUE TITLE
calc_WodaFuchs2008() crashes on some RLum.Results files

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#199]
For example:
```R
data(ExampleData.DeValues)
res <- calc_WodaFuchs2008(data = ExampleData.DeValues$CA1, breaks = 10)
calc_WodaFuchs2008(res)

# Error: [get_RLum()] data.object(s) unknown, valid names are: D_estimate, breaks
```

There should be some checks on the originator, but which are the valid ones?

**Comment by mcol** [comment]
@RLumSK I need your input on the valid originators to make progress on this.

**Comment by RLumSK** [comment]
It can be `data <- get_RLum(data)` because by convention the default element in the data slot should be the most meaningful.

I had tested it quickly locally, the function will then somehow stuck in an infinite loop because the data frame has only one row. Perhaps we should also add a check for this or avoid that `nls()` will later stuck.

**Comment by mcol** [comment]
In this case, given that there is just one point, the computation of the histogram produces just one bin, so `counts_log` has length 1. But the computation of the curvature expects at least 3 points:
https://github.com/R-Lum/Luminescence/blob/759a1142651cc557f9fd06c6b061d1aff244cdf9/R/calc_WodaFuchs2008.R#L155-L157
In my example, I get `curvature = NA`.

With two points, the code generates some infinities, which result in `curvature = NaN`:
```R
H <- hist(x = c(data[,1], 60), breaks = 4, plot = FALSE)
print(log(H$counts))
# [1]    0 -Inf -Inf -Inf    0
```
A similar thing happens with 3 points. With 4 points it may fail or not, depending on how the values get binned.

It seems that rather than setting a minimum number of points, we should just see if `curvature` gets a proper value and return with an error if that's not the case.

**Comment by mcol** [comment]
I just found an example in #197 in which the curvature is `-Inf`:
```R
data(ExampleData.Fading, envir = environment())
df <- ExampleData.Fading$fading.data$IR50
calc_WodaFuchs2008(df)
```
This has 24 points, but one of the histogram bins is empty.

The safer would be to check that we have at least 3 bins, and none have 0 counts.

**Comment by mcol** [comment]
The more I look at this, the less it makes sense:

- The curvature will be infinite if the 1st and 3rd bin have the same count (which is not a remote possibility) as it divides by `(counts_log[1] - counts_log[3])`
- It's unclear why we should only use the first 3 elements of `count_log`, as if the user sets a large number of histogram breaks, then `count_log` will be much longer than 3
- At the end of the day, it doesn't quite matter as `curvature` is never used anywhere in the code after it's computed!
- There is a minimum number of breaks set at 4 which seems somewhat arbitrary, in particular if the curvature computation is unused
- I've turned on tracing and added `control = list(maxiter = 1, printEval = TRUE)` to the `nls()` call and in the original example it prints nothing, meaning that it doesn't even complete a single iteration (I waited about 2 minutes), suggesting that perhaps there's something quite unexpected happening inside the C code that implements the "port" algorithm (it gets called from `stats:::nls_port_fit()`)

In order to fix the original example of the issue without breaking any of the existing tests, it turns out that the best approach is just to return if there is only one data point. This is a bit unsatisfactory but it will do for this release, in any case this function deserves a better look to make it more reliable.
